feat: 배송 상세 조회 API 구현

Jungdaye89 opened the pull request:
## 개요
- 배송 ID로 배송 상세 정보를 조회할 수 있는 API를 구현했습니다.  
- 배송 정보와 주문 정보를 통합하여 하나의 응답으로 내려주는 구조입니다.

<br>

## 📝 작업 내용
### 변경 사항
- 배송 도메인
  - `DeliveryRepositoryAdapter`에 `findByDeliveryId()` 메서드 추가 및 예외 처리 (`DELIVERY_NOT_FOUND`)
  - `DeliveryService`에서 배송 정보 조회 → 주문 API 호출 → 응답 DTO 변환까지 전체 흐름 구현
  - 배송 상세 응답용 DTO (`DeliveryDetailResponseDto`) 정의
  - 배송 상세 컨트롤러 API (`GET /api/v1/deliveries/{deliveryId}`) 추가
  - `DeliveryResponseMapper`를 통해 배송 + 주문 정보를 통합 응답 DTO로 매핑

- OrderClient
  - 주문 서비스와 연동되는 `OrderClient` 구현 (`FeignClient`)
  - 주문 API 응답을 `Map<String, Object>` 형태로 받아 필요한 필드만 추출
  - 주문 요약 DTO (`OrderInfoResponseDto`) 생성 및 응답에 포함

### 변경 이유

### 추가 설명

<br>

### 💬리뷰 요구사항

<br>

### #️⃣ 연관된 이슈
closed #193 

<br>

## PR 유형

이 PR은 어떤 변경 사항을 포함하고 있나요?

- [X] 새로운 기능 추가
- [ ] 버그 수정
- [ ] UI 디자인 변경 (CSS 등)
- [ ] 코드에 영향을 주지 않는 변경사항 (오타 수정, 탭 사이즈 변경, 변수명 변경 등)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 또는 폴더명 수정
- [ ] 파일 또는 폴더 삭제
- [ ] 배포 준비, PR 관련 사항

<br>

## ✅ Check List

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [X] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다. (버그 수정/기능에 대한 테스트)
- [ ] CI/CD 파이프라인을 통과했습니다.

Jungdaye89 committed:
feat: 배송 상세 조회 API

- /api/v1/deliveries/{deliveryId} 경로로 배송 + 주문 정보 통합 조회 제공

Resolved #193

Jungdaye89 committed:
feat: 배송 상세 조회 비즈니스 로직 구현

- 배송 정보 조회 → 주문 API 호출 → 매퍼를 통해 DTO로 변환하는 전체 로직 구현

Resolved #193

Jungdaye89 committed:
feat: 배송 + 주문 정보 응답 DTO 매핑 로직 추가

- 주문 응답 Map<String, Object>에서 필요한 필드 추출하여 DeliveryDetailResponseDto로 변환

Resolved #193

Jungdaye89 committed:
feat: 배송 상세 조회 응답 DTO 생성

- 배송 정보, 배송 경로 리스트, 주문 정보 포함하는 상세 응답 DTO 설계 및 필드 정의

Resolved #193

Jungdaye89 committed:
feat: 주문 정보를 담는 DTO 생성

- 가지고 오고 싶은 주문 정보 담는 DTO

Resolved #193

Jungdaye89 committed:
feat: findByDeliveryId 추가

Resolved #193

Jungdaye89 committed:
feat: findByDeliveryId 구현체 작성 및 예외 처리 추가

- 조회 실패 시 BusinessLogicException으로 DELIVERY_NOT_FOUND 에러 던지도록 구현

Resolved #193

Jungdaye89 committed:
feat: 배송 ID 조회 실패 시 사용될 에러 메시지 정의

Resolved #193

Jungdaye89 committed:
feat: 주문 조회 FeignClient 인터페이스 추가

- /api/v1/orders/{orderId} 호출하여 주문 정보를 Map<String, Object>로 반환하도록 정의

Resolved #193

Jungdaye89 committed:
refactor: import 정리

Resolved #193

dowon0113 commented on delivery/src/main/java/com/business/delivery/infrastructure/client/OrderClient.java:
```diff
@@ -0,0 +1,14 @@
+
+    @GetMapping("/api/v1/orders/{orderId}")
+    Map<String, Object> getOrder(@PathVariable("orderId") UUID orderId);
+}
```
dto가 아닌 Map<String, Object>로 받는 방식도 있었네요! 참고하겠습니다~!

AnJungGeon replied:
DTO는 데이터 전달 객체라는 의미를 가지므로, 전달하고자 하는 데이터가 명확한 구조를 가질 수 있다면 Map보다는 객체로 정의하는 것이 바람직합니다 😄

Jungdaye89 replied:
현재 주문 애그리거트 내부의 주문 아이템 정보 중 필요한 값만 추출하기 위해 Map을 사용했었습니다. 리팩토링 시에는 Map 대신 DTO 객체로 데이터를 매핑하는 구조로 적용해보겠습니다 :)

Jungdaye89 committed:
Merge branch 'develop' into feat/delivery-read

Jungdaye89 committed:
refactor: 파일 이름 변경

Resolved #172

Jungdaye89 committed:
refactor: DeliveryPageListResponseDto의 fromPage 메서드를 DeliveryResponseMapper로 분리

- 페이지 응답 변환 로직의 책임 분리를 위해 fromPage 메서드를 DeliveryResponseMapper로 이동
- DeliveryPageListResponseDto는 순수 데이터 구조만 유지하도록 개선
- DeliveryController 및 서비스 단에서 변환 시 DeliveryResponseMapper.toPageListResponse() 호출하도록 변경

Resolved #172
